### PR TITLE
[chip-tool] Rename executable to chip-tool and improve quick start instructions.

### PR DESCRIPTION
**Pull request by turon**
#### Problem
The chip-tool code outputs an executable named `chip-standalone-demo`.

 #### Summary of Changes
Rename the executable to simply `chip-tool`.

### CODE REVIEW

**Review by BroderickCarlin — APPROVED** [review]

**Comment by woody-apple** [comment]
@bzbarsky-apple ?

@turon also, needs merge conflict fixes :(

File: examples/chip-tool/README.md [hunk 1]
```diff
@@ -14,20 +14,41 @@ An example application that uses CHIP to send messages to a CHIP server.
 
 Building the example application is quite straightforward.
 
+### autotools
+
+- From the top-level of the chip repo, run `./gn_build.sh`
```
[the remark this answers is not in the document]
Comment (turon) [reply examples/chip-tool/README.md:19]: Added your instructions in addition to general mechanism.

**Comment by turon** [comment]
Restyled seems to be getting confused.  `origin/restyled/pull-2292` branch doesn't seem to exist...

```
stderr ! [remote rejected]   restyled/pull-2292 -> restyled/pull-2292 (refusing to allow a GitHub App to create or update workflow `.github/workflows/examples.yaml` without `workflows` permission)
stderrerror: failed to push some refs to 'https://<SCRUBBED>@github.com/project-chip/connectedhomeip.git'
stderrWe had trouble running a system command:
stderr
stderr  callProcess: git "push" "--force-with-lease" "origin" "restyled/pull-2292" (exit
stderr  1): failed
stderr
stderr
systemRestyler exited 50
```

**Comment by rwalker-apple** [comment]
> Restyled seems to be getting confused. `origin/restyled/pull-2292` branch doesn't seem to exist...
> 
> ```
> stderr ! [remote rejected]   restyled/pull-2292 -> restyled/pull-2292 (refusing to allow a GitHub App to create or update workflow `.github/workflows/examples.yaml` without `workflows` permission)
> stderrerror: failed to push some refs to 'https://<SCRUBBED>@github.com/project-chip/connectedhomeip.git'
> stderrWe had trouble running a system command:
> stderr
> stderr  callProcess: git "push" "--force-with-lease" "origin" "restyled/pull-2292" (exit
> stderr  1): failed
> stderr
> stderr
> systemRestyler exited 50
> ```

The above is an issue with restyled, where it really wants to help, but because your PR touches a file it can't include in a PR, it falls over.

To restyle locally (from the chip root):
```
$ curl https://raw.githubusercontent.com/restyled-io/restyler/master/bin/restyle-path= > restyle-path
$ chmod 755 restyle-path
$ RESTYLER_TAG=:master-final ./restyle-path <files in this PR>
```
Running restyle-path from the chip root is important because that's where the config lives...

**Review by rwalker-apple — APPROVED** [review]
I did the restyle thing, too

**Review by mspang — APPROVED** [review]

**Pull request merged**